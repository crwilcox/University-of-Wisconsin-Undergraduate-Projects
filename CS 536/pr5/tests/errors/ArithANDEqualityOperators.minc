void main() {
int I = 0;
double D = 1.0;
string S = "";
bool B = true;
int* P;

// operator (%) may only be applied to arguments of type int.
int Modulo;
Modulo = I % D;
Modulo = I % S;
Modulo = I % B;
Modulo = I % P;

int Num;
Num = B + B;
Num = S + S;
Num = P + P;

Num = B - B;
Num = B - B;
Num = B - B;

Num = B * B;
Num = B * B;
Num = B * B;

Num = B / B;
Num = S / S;
Num = P / P;

Num = S < S;
Num = P < P;
Num = S < S;

Num = P > P;
Num = S > S;
Num = P > P;

Num = S <= S;
Num = P <= P;
Num = S <= S;

Num = P >= P;
Num = S >= S;
Num = P >= P;
}
