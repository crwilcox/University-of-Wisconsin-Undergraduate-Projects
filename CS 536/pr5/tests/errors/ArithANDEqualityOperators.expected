10:12:Operator can only be applied to numeric types
11:12:Operator can only be applied to numeric types
12:12:Operator can only be applied to numeric types
13:12:Operator can only be applied to numeric types
16:9:Operator can only be applied to numeric types
17:9:Operator can only be applied to numeric types
18:9:Operator can only be applied to numeric types
20:9:Operator can only be applied to numeric types
21:9:Operator can only be applied to numeric types
22:9:Operator can only be applied to numeric types
24:9:Operator can only be applied to numeric types
25:9:Operator can only be applied to numeric types
26:9:Operator can only be applied to numeric types
28:9:Operator can only be applied to numeric types
29:9:Operator can only be applied to numeric types
30:9:Operator can only be applied to numeric types
32:9:Operator can only be applied to numeric types
32:5:Type mismatch
33:9:Operator can only be applied to numeric types
33:5:Type mismatch
34:9:Operator can only be applied to numeric types
34:5:Type mismatch
36:9:Operator can only be applied to numeric types
36:5:Type mismatch
37:9:Operator can only be applied to numeric types
37:5:Type mismatch
38:9:Operator can only be applied to numeric types
38:5:Type mismatch
40:9:Operator can only be applied to numeric types
40:5:Type mismatch
41:9:Operator can only be applied to numeric types
41:5:Type mismatch
42:9:Operator can only be applied to numeric types
42:5:Type mismatch
44:9:Operator can only be applied to numeric types
44:5:Type mismatch
45:9:Operator can only be applied to numeric types
45:5:Type mismatch
46:9:Operator can only be applied to numeric types
46:5:Type mismatch
